modules/analytics/src/views/full/TimeSeriesChart.tsx

```diff
@@ -47,10 +40,10 @@ const TimeSeriesChart: FC<Props> = props => {
                   .filter(x => x.value !== 'all')
                   .map(({ value: channel }, idx) => (
                     <linearGradient key={idx} id={`gradientBg-${channel}`} x1="0" y1="0" x2="0" y2="1">
-                      <stop offset="0%" stopColor={CHANNEL_COLORS[channel] || '#000'} stopOpacity={0.31} />
-                      <stop offset="45%" stopColor={CHANNEL_COLORS[channel] || '#000'} stopOpacity={0.34} />
-                      <stop offset="73%" stopColor={CHANNEL_COLORS[channel] || '#000'} stopOpacity={0.15} />
-                      <stop offset="100%" stopColor={CHANNEL_COLORS[channel] || '#000'} stopOpacity={0} />
+                      <stop offset="0%" stopColor={CHANNEL_COLORS[idx] || '#000'} stopOpacity={0.31} />
+                      <stop offset="45%" stopColor={CHANNEL_COLORS[idx] || '#000'} stopOpacity={0.34} />
+                      <stop offset="73%" stopColor={CHANNEL_COLORS[idx] || '#000'} stopOpacity={0.15} />
+                      <stop offset="100%" stopColor={CHANNEL_COLORS[idx] || '#000'} stopOpacity={0} />
```

Comment:
If you have more than 5 channels, you will get an out of bound error. Maybe use a method instead of direct assignment ?

Reply:
@allardy Yes. That's better.